codalab_service.py

```diff
@@ -697,6 +691,7 @@ def start_services(self):
                     ('cli/default_address', rest_url),
                     ('server/engine_url', mysql_url),
                     ('server/rest_host', '0.0.0.0'),
+                    ('server/rest_port', self.args.rest_port),
```

Comment:
This took me a long time to figure out, but I noticed that we never respected the value of the `rest_port` parameter that is being passed in and just kept it hardcoded at the default value of `2900`.